File: Go/058. Length of Last Word.go
package main

func lengthOfLastWord(s string) int {
	var i, j int
	for j = len(s) - 1; j >= 0 && s[j] == ' '; j-- {}
	for i = j; i >= 0 && s[i] != ' '; i-- {}
	return j - i
}

File: C++/058. Length of Last Word.cpp
#include <iostream>

class Solution {
public:
    int lengthOfLastWord(std::string s) {
        int i, j;
        for (j = int(s.size()) - 1; j >= 0 && s[j] == ' '; j--);
        for (i = j; i >= 0 && s[i] != ' '; i--);
        return j - i;
    }
};